stdlib/public/core/StringProtocol.swift

```diff
@@ -177,15 +177,27 @@ extension StringProtocol {
 
 // Contiguous UTF-8 strings
 extension String {
-  /// Returns whether this string is capable of providing access to
-  /// validly-encoded UTF-8 contents in contiguous memory in O(1) time.
+  /// Returns whether this string's storage contains
+  /// validly-encoded UTF-8 contents in contiguous memory.
   ///
-  /// Contiguous strings always operate in O(1) time for withUTF8 and always
-  /// give a result for String.UTF8View.withContiguousStorageIfAvailable.
+  /// Contiguous strings always operate in O(1) time for withUTF8, always give
+  /// a result for String.UTF8View.withContiguousStorageIfAvailable, and always
```

Comment:
Was this comment actually inaccurate before? Did `String.UTF8View.withContiguousStorageIfAvailable` actually NOT provide a result for these exceptional small strings?

Reply:
It was not lying: `UTF8View.withContiguousStorageIfAvailable` will rearrange the bytes of a small string in order to present contiguous storage to the closure.

Reply:
OK. With this returning `false` now for these strings, some might find it surprising that the former conditions are still true. However, seems like we're within our rights to do so, as this is implication ("isContiguousUTF8 -> the above guarantees hold") and not equivalence ("isContiguousUTF8 <-> the above guarantees hold").